premarum-backend/PreEnrollmentMgmt.Core/Entities/Student.cs

```diff
@@ -28,4 +28,9 @@ public void AddPreEnrollment(PreEnrollment preEnrollment)
     {
         PreEnrollments.Add(preEnrollment);
     }
+
+    public void AddCoursesTaken(CoursesTaken coursesTaken)
```

Comment:
Instead of sending a CoursesTaken object, send the parameters for the CoursesTaken constructor on this function.